packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -87,6 +119,42 @@ export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
     }
 
     public dispose() {
-        this.idleTimer.clear();
+        this.idleTimer?.clear();
     }
 }
+
+export class MaxTimeSummarizeHeuristicStrategy implements ISummarizeHeuristicStrategy {
+    public readonly summarizeReason: Readonly<SummarizeReason> = "maxTime";
+
+    public shouldRunSummarize(configuration: ISummaryConfiguration, heuristicData: ISummarizeHeuristicData): boolean {
```

Comment:
>shouldRunSummarize [](http://example.com/codeflow?start=11&length=18)

nit: would it be better shouldRunSummarizer ?

Reply:
I agree that the naming of these isn't the nicest. I was following the current naming scheme since the entry method is called `trySummarize` and other things are named similarly (ex: `SummarizeReason`)